src/Blueprint/Blueprint/Cmdlets/BlueprintCmdletBase.cs

```diff
@@ -0,0 +1,382 @@
+﻿using Microsoft.Azure.Commands.Blueprint.Common;
+using Microsoft.Azure.Commands.Common.Authentication;
+using Microsoft.Azure.Commands.Common.Authentication.Abstractions;
+using Microsoft.Azure.Commands.ResourceManager.Common;
+using Microsoft.Azure.Management.ManagementGroups;
+using Microsoft.Azure.Internal.Subscriptions;
+using System;
+using Microsoft.Rest;
+using Microsoft.Rest.Azure;
+using Microsoft.Azure.PowerShell.Cmdlets.Blueprint.Properties;
+using Microsoft.Azure.Commands.Blueprint.Models;
+using Microsoft.Azure.Management.Blueprint.Models;
+using System.Collections.Generic;
+using System.Collections;
+using System.Linq;
+using System.Net.Http;
+using System.Threading;
+using Microsoft.Azure.PowerShell.Cmdlets.Blueprint.Common;
+using Newtonsoft.Json.Linq;
+using System.Threading.Tasks;
+using Microsoft.Azure.Management.Internal.ResourceManager.Version2018_05_01;
+using Microsoft.Azure.Graph.RBAC.Version1_6;
+using Microsoft.Azure.Graph.RBAC.Version1_6.Models;
+using Microsoft.Azure.Management.Authorization.Version2015_07_01;
+using Microsoft.Azure.Management.Authorization.Version2015_07_01.Models;
+
+namespace Microsoft.Azure.Commands.Blueprint.Cmdlets
+{
+    public class BlueprintCmdletBase : AzureRMCmdlet
+    {
+        #region Properties
+        /// <summary>
+        /// The blueprint client.
+        /// </summary>
+        private IBlueprintClient blueprintClient;
+
+        public IBlueprintClient BlueprintClient
+        {
+            get
+            {
+                return blueprintClient = blueprintClient ?? new BlueprintClient(DefaultProfile.DefaultContext);
+            }
+            set => blueprintClient = value;
+        }
+
+        /// <summary>
+        /// The blueprint client with delegating handler. The delegating handler is needed to get blueprint versions info.
```

Comment:
>The delegating handler is needed to get blueprint versions info. [](start = 58, length = 64)

We should probably file a task to model $expand in the swagger or via some SDK customization so it's easy for users of the SDK to do it.

Reply:
Created a task. [https://msazure.visualstudio.com/One/_workitems/edit/3926599](https://msazure.visualstudio.com/One/_workitems/edit/3926599) 


---
In reply to: [257091752](https://github.com/Azure/azure-powershell/pull/8537#discussion_r257091752) [](ancestors = 257091752)